compiler/rustc_infer/src/infer/resolve.rs

```diff
@@ -138,11 +138,13 @@ impl<'a, 'tcx> FallibleTypeFolder<TyCtxt<'tcx>> for FullTypeResolver<'a, 'tcx> {
         if !t.has_infer() {
             Ok(t) // micro-optimize -- if there is nothing in this type that this fold affects...
         } else {
+            use super::TyOrConstInferVar::*;
```

Comment:
While you're at it, can you update the comment on `FullTypeResolver`?
```rust
// N.B. This type is not public because the protocol around checking the
// `err` field is not enforceable otherwise.
```
this does not make sense anymore